### PR TITLE
ENH: add color table and terminology for DICOM segmentation reading

### PR DESCRIPTION
**Pull request by naucoin**
When reading in a DICOM segmentatin file, create one color table node
for all the segments, and build up a terminology associated with the
segmentation from the .info files.
Adjusted the logic call to not use the color node, do the terminology and color
information file parsing in the segmentation plugin.
Added a check that the SEG2NRRD CLI was loaded to prevent a crash if Slicer was started up
without CLI modules.

Issue #52

### CODE REVIEW

**Comment by pieper** [comment]
Looks good to me.  What DICOM segmentation objects do you use for testing this?

**Comment by fedorov** [comment]
Very good! Is the Slicer part ready for integration, once the freeze is lifted?

**Comment by fedorov** [comment]
For the reference, this is the branch of Slicer with which this is supposed to work with: https://github.com/naucoin/Slicer/tree/4047-Add-semantic-color-information-to-color-selector-in-editor

**Comment by naucoin** [comment]
Andrey sent me a segmentation object called tumor_User1_Manual_Trial1.dcm with 10 segmentations, here's a screen shot of most of the color node I built from it:
(https://cloud.githubusercontent.com/assets/289054/10971578/fe8860b0-83a2-11e5-8b70-a51b2f090753.jpg)

**Comment by naucoin** [comment]
@fedorov The display in the editor part is ready for integration, I'm continuing work on parsing the xml files to allow editing/setting new terminology.

**Comment by fedorov** [comment]
Nicole, let me rephrase my question. If I compile your branch of Slicer, and try to use this updated version of the Reporting module, should I expect it to work and link the labels with colors and terminology correctly?

**Comment by naucoin** [comment]
Yes.

**Comment by naucoin** [comment]
I have an update to DataProbe.py that shows the terminology on mouse over, but I haven't checked that in, so the currently shared version of Slicer has the terminology under the hood for developers. If you have a label integer that you want terminology for, you can print it:
colorLogic = slicer.modules.colors.logic()
segLut = getNode('vtkMRMLColorTableNode1')
colorLogic.PrintCategorizationFromLabel(1, segLut.GetName())
or get the terminology strings:
colorLogic.GetCategoryFromLabel(1, segLut.GetName())
colorLogic.GetCategoryTypeFromLabel(1, segLut.GetName())
colorLogic.GetCategoryModifierFromLabel(1, segLut.GetName())
